internal/native_image/builder.bzl

```diff
@@ -64,15 +154,21 @@ def assemble_native_build_options(
         direct_inputs: Direct inputs into the native image build (mutable).
         c_compiler_path: Path to the C compiler; resolved via toolchains.
         path_list_separator: Platform-specific path separator.
-        gvm_toolchain: Resolved GraalVM toolchain, or `None` if a tool target is in use via legacy rules. """
+        gvm_toolchain: Resolved GraalVM toolchain, or `None` if a tool target is in use via legacy rules.
+    """
 
-    args.add("--no-fallback")
+    # main class is required unless we are building a shared library
+    if ctx.attr.shared_library:
+        args.add("--shared")
```

Comment:
note to self: add `native_image_shared_library` as alias which selects header outputs automatically in addition to binary shared lib